checkEffects.R

```diff
@@ -1050,4 +1048,115 @@ myeff$test <- FALSE
 (ans2 <- siena07(mymodel, data=mydata, effects=myeff, prevAns=ans1))
 (ans3 <- siena07(mymodel, data=mydata, effects=myeff, prevAns=ans2))
 
+################################################################################
+### check sharedTo
+################################################################################
 
+mynet1 <- sienaNet(array(c(s501, s502), dim=c(50, 50, 2)))
+mynet2 <- sienaNet(array(c(s502, s503), dim=c(50, 50, 2)))
+mydata <- sienaDataCreate(mynet1, mynet2)
+mymodel <- sienaModelCreate(projname=NULL, seed=1234)
+myeff <- getEffects(mydata)
+myeff <- setEffect(myeff, sharedTo, name="mynet2", interaction1="mynet1", parameter=1)
+(ans <- siena07(mymodel, data=mydata, effects=myeff))
+ans$targets
+# 115 116  70 106 122  90 154
+intwostars1 <- (s501) %*% t(s501)
+intwostars3 <- (s503) %*% t(s503)
+diag(intwostars1) <- 0
+diag(intwostars3) <- 0
+sum(intwostars1 * intwostars3) # OK
+
+###
+myeff <- getEffects(mydata)
+myeff <- setEffect(myeff, sharedTo, name="mynet2", interaction1="mynet1", parameter=2)
+(ans <- siena07(mymodel, data=mydata, effects=myeff))
+ans$targets
+# 115 116  70 106 122  90 139.7484
+sum(sqrt(intwostars1) * intwostars3) # OK
+
```

Comment:
I am not totally happy with this way of testing effects, but currently I'm not aware of another easy way. The lack of this test is its limitation to the estimation statistic, ignoring the change statistic (except that convergence is some signal, albei limited, of correspondence between the two).
Incidentally, the error in the sharedTo effect was that, when I first tested it (7 years ago) I used a data set for which the incorrect estimation statistic provided the correct numerical answer, which misled me.
Earlier you (James) had included in RSiena a dependence on testthat, but I did not see that being operational and therefore dropped it. I'm reluctant to include dependencies on other packages, because that leads to extra issues when those packages are changed. 
When one of the team members proposes and implements a further way of checking, which might use testthat or tinytest, that will be fine and useful; but only when at least some meaningful things are indeed operational.

Reply:
We could open this as a new issue or discussion on Github to get feedback/ideas in the medium term. Indeed moving to a new testing structure would be a major time investment.